src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -0,0 +1,667 @@
+        {
+            var employeesIds = new JsonArray();
+
+            foreach (var employee in EmployeesDatabase.GetTenBestEmployees())
+            {
+                employeesIds.Add(employee.Key);
+            }
+        }
+
+        /// <summary>
+        /// Creating Json array from collection
+        /// </summary>
+        [Fact]
+        public static void TestCreatingJsonArrayFromCollection()
+        {
+            var employeesIds = new JsonArray(EmployeesDatabase.GetTenBestEmployees().Select(employee => new JsonString(employee.Key)));
+        }
+
+        /// <summary>
+        /// Creating Json array from collection of strings
+        /// </summary>
+        [Fact]
+        public static void TestCreatingJsonArrayFromCollectionOfString()
+        {
+            var employeesIds = new JsonArray(EmployeesDatabase.GetTenBestEmployees().Select(employee => employee.Key));
+        }
+
+        /// <summary>
+        /// Contains checks
+        /// </summary>
+        [Fact]
+        public static void TestContains()
+        {
+            var person = new JsonObject
+            {
+                { "name", "John" },
+                { "ssn", "123456789" },
+            };
+
+            if (person.ContainsProperty("ssn"))
+            {
+                EmployeesDatabase.CheckSSN(((JsonString)person["ssn"]).GetString());
+            }
+
+            var enabledOptions = new JsonArray
+            {
+                "readonly",
+                "no cache",
+                "continue on failure"
+            };
+
+            if (enabledOptions.Contains("no cache"))
+            {
+                // do sth without using caching
+            }
+
+            var requiredOptions = new JsonArray
+            {
+                "readonly",
+                "continue on failure"
+            };
+
+            // if all required options are enabled
+            if (!requiredOptions.Select(option => !enabledOptions.Contains(option)).Any())
+            {
+                // do sth without using caching
+            }
+        }
+
+        /// <summary>
+        /// Modifying Json object's primnary types
+        /// </summary>
+        [Fact]
+        public static void TestModifyingJsonObjectPrimaryTypes()
+        {
+            var person = new JsonObject
+            {
+                { "name", "John" },
+                { "age", 45 },
+                { "is_married", true }
+            };
+
+            // Assign by creating a new instance of primary Json type
+            person["name"] = new JsonString("Bob");
+
+            // Assign by using an implicit operator on primary Json type
+            JsonNumber newAge = 55;
+            person["age"] = newAge;
+
+            // Assign by explicit cast from Json primary type
+            person["is_married"] = (JsonBool)true;
+
+            // Not possible scenario (wold require implicit cast operators in JsonNode):
+            // person["name"] = "Bob";
```

Comment:
Do we want to use implicit cast operators? Currently they are added only for `JsonString`, `JsonNumber` and `JsonBool` as they seems to be helpful in creating instances of those types. But for example this line wouldn't work without implicit cast operator in `JsonNode` itself.

Reply:
The only way to define the operator is to say that String can turn into JsonNode, but it returns a predictably more-derived type: JsonString.

While I agree adding them would let fewer words be written, I don't know that the brevity is valuable... so I'd advise waiting for a specific request for it.